qutebrowser/browser/hints.py

```diff
@@ -118,10 +118,15 @@ def update_text(self, matched: str, unmatched: str) -> None:
             matched: The part of the text which was typed.
             unmatched: The part of the text which was not typed yet.
         """
-        if (config.cache['hints.uppercase'] and
+        if (config.cache['hints.labels'] and
                 self._context.hint_mode in ['letter', 'word']):
-            matched = html.escape(matched.upper())
-            unmatched = html.escape(unmatched.upper())
+            labels = config.val.hints.labels
+            chars = config.val.hints.chars
+            label_dict = {}
+            for i in range(min(len(chars), len(labels))):
+                label_dict[chars[i]] = labels[i]
```

Comment:
Thanks for the feedback! 

I think it would be (marginally) useful to let the `chars` and `labels` be of different lengths without showing errors. For example by allowing different lengths `labels = '←↓↑→↖↗↙↘'` can be used with both `chars = 'hjklyunb'` and `chars = 'hjkl'`. This also applies if a user wants to use, for example, numpad with arrows with or without diagonal directions. Maybe there are other use cases?

I wonder if this is a good way to go about this and what are the downsides to letting the strings be of different lengths?

Reply:
I made the change to use `zip` as per your first suggestion, but I'll leave this conversation open for now, if you want to pursue the second option.